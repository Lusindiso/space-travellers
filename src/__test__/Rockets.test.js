import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rockets from '../components/Rockets';
// import rocketReducer from '../redux/rockets/rockets';

describe('Rocket Tests', () => {
  test('for proper rendering of rockets', () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(screen.findAllByRole('button', { name: /Reserve Rocket/ }).toMatchSnapshot);
  });

  it('renders porperly', () => {
    const rocket = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(rocket).toMatchSnapshot();
  });
});
