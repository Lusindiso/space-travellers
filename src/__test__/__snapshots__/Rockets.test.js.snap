// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket Tests renders porperly 1`] = `
<div
  className="row"
>
  <hr />
</div>
`;
